packages/utils/src/stacktrace.ts

```diff
@@ -1,3 +1,69 @@
+import { StackFrame } from '@sentry/types';
+
+const STACKTRACE_LIMIT = 50;
+
+export type StackLineParser = (line: string) => StackFrame | undefined;
+
+/**
+ * Creates a stack parser with the supplied line parsers
+ *
+ * StackFrames are returned in the correct order for Sentry Exception
+ * frames and with Sentry SDK internal frames removed from the top and bottom
+ *
+ * */
+export function createStackParser(...parsers: StackLineParser[]) {
+  return (stack: string, skipFirst: number = 0): StackFrame[] => {
```

Comment:
Why choose to do the higher order function and curry the args instead of just explicitly passing in an array of parsers?

I can't see any need for this to be lazy since we don't do any incremental computations on the parsers.

Reply:
In a future PR, the browser/node SDKs are going to end up with configurable stack parsing. Encapsulating like this seemed better than passing around an array of line parsers which need an additional function importing when you want to do anything with them.

Reply:
We crossed paths and I missed your edit 😃